lib/mina/local_helpers.rb

```diff
@@ -64,16 +61,18 @@ def invoke(script, this)
         code = "#{script}"
 
         # Certain environments can't do :pretty mode.
-        term_mode = :exec  if term_mode == :pretty && !pretty_supported?
+        term_mode = :exec if term_mode == :pretty && !pretty_supported?
 
         case term_mode
         when :pretty
+          require 'shellwords'
+          code = Shellwords.escape(code)
           this.pretty_system(code)
         when :exec
-          exec code
+          Kernel.exec code
```

Comment:
This will completely replace the ruby process with a shell running `code`. Not sure if it's the expected behavior.